lib/screenplay/config/permanent_config.ex

```diff
@@ -196,15 +208,6 @@ defmodule Screenplay.Config.PermanentConfig do
       else: {:error, {:duplicate_screen_ids, duplicate_screen_ids}}
   end
 
-  defp get_updated_pending_screen_ids(updated_pending_screens) do
-    Enum.filter(updated_pending_screens, fn screen ->
-      not is_nil(screen) and Map.has_key?(screen, "new_id")
```

Comment:
I think we still need to keep this filter somewhere in the pipeline, because otherwise when we map out `"new_id"` above, we could get nils, which would mess up the later logic.

Reply:
I made an attempt to improve the algorithm in the function that was calling this. It should more clearly group IDs based on the type of change. That should allow a guarantee that we won't see `nil` values when performing our checks.